test/WebJobs.Extensions.RabbitMQ.Tests/RabbitMQServiceTests.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using Microsoft.Azure.WebJobs.Extensions.RabbitMQ;
+using Moq;
+using RabbitMQ.Client;
+using Xunit;
+
+namespace WebJobs.Extensions.RabbitMQ.Tests
+{
+    public class RabbitMQServiceTests
+    {
+        [Theory]
+        [InlineData(null, "localhost", "queue", "guest", "guest", 5672)]
+        [InlineData("amqp://user:PASSWORD@52.175.195.81:5672", null, "queue", null, null, null)]
+        [InlineData(null, "localhost", "queue", null, null, 0)] // Should fill in "guest", "guest", 5672
+        public void Handles_Connection_Attributes_And_Options(string connectionString, string hostName, string queueName, string userName, string password, int port)

```

Comment:
pass expected values as part of params. Yes signature will look ugly, but test code will lot simpler as you just need to check for expected values